### DISCUSSION TITLE
Shadowsocks Transproxy

### DISCUSSION THREAD

**Discussion by MasterKia** [celzero/rethink-app#675] [Q&A]
What is this transproxy and how is it different from regular SOCKS5 proxy?
Does Rethink support the transproxy that Shadowsocks provides?

![IMG_20221118_090139_466](https://user-images.githubusercontent.com/17685483/202628065-0020b3e1-38c9-4b1e-ad02-1e387592ab1a.jpg)
![IMG_20221118_090137_600](https://user-images.githubusercontent.com/17685483/202628070-1c35ba43-df1a-4630-8623-1536c2a75e52.jpg)

**Comment by MasterKia** [comment]
https://github.com/shadowsocks/shadowsocks-android/blob/master/.github/faq.md#how-to-use-transproxy-mode

**Reply by ignoramous** [reply]
From a cursory look, it seems like *transproxy* is essentially routing traffic through shadowsocks using `iptables` / `nftables`. That is, the apps wouldn't even know there's shadowsocks that's doing the routing. I guess, the proxy appears "transparent" to unsuspecting apps.

Amusingly, the example in the readme you shared doesn't use the *actual* `TPROXY` module: https://serverfault.com/questions/1056072

---

As for Rethink, I think the shadowsocks transproxy option should work if it really is transparent... If it doesn't, then it is a question for whoever wrote the readme.
